### PR TITLE
fix(BottomSheet): fallback to background.default and adapt text color

### PR DESCRIPTION
**Pull request by KhushamBansal**
**Notes for Reviewers**

This PR fixes #

Screenshot 2026-08-28 at 8 57 58 PM

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Yes, I signed my commits.

## Summary by CodeRabbit

* **New Features**
  * Added customizable header background and text colors for dashboard layouts and mobile bottom sheets.
  * Bottom Sheet headers now use the theme’s surface tint when no custom background is provided.

* **Bug Fixes**
  * Improved header text and close-button contrast across light and dark themes.
  * Corrected fallback behavior for empty or unavailable header backgrounds.
  * Updated default header colors to align with application theme settings.

### CODE REVIEW

**Review by PARTH-TUSSLE — CHANGES_REQUESTED** [review]
Hey @KhushamBansal , the main thing I noticed is that the text/icon color fallback is checking whether surface.tint exists, rather than the background that actually ends up being rendered.
For example, if someone passes headerBackgroundColor="#fff", we'd still get common.white for the text because surface.tint exists, which gives us white-on-white.
I think we should resolve the final header background first, then derive the default text color from that, while still letting an explicit headerTextColor override it. Also, we could resolve that text color once and reuse it for both the title and close icon instead of duplicating the logic.

**Comment by KhushamBansal** [comment]
> Hey @KhushamBansal , the main thing I noticed is that the text/icon color fallback is checking whether surface.tint exists, rather than the background that actually ends up being rendered. For example, if someone passes headerBackgroundColor="#fff", we'd still get common.white for the text because surface.tint exists, which gives us white-on-white. I think we should resolve the final header background first, then derive the default text color from that, while still letting an explicit headerTextColor override it. Also, we could resolve that text color once and reuse it for both the title and close icon instead of duplicating the logic.


@PARTH-TUSSLE Thanks for reviewing this PR. I have made the change.

File: src/custom/BottomSheet/BottomSheet.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import Slide, { SlideProps } from '@mui/material/Slide';
+import { useTheme } from '@mui/material/styles';
```
Comment (NSTKrishna) [inline src/custom/BottomSheet/BottomSheet.tsx]: use it from ../../theme

**Review by NSTKrishna — CHANGES_REQUESTED** [review]

File: src/custom/BottomSheet/BottomSheet.tsx [hunk 2]
```diff
@@ -38,6 +39,14 @@ const BottomSheet = ({
...
 }: BottomSheetProps) => {
   const titleId = useId();
+  const theme = useTheme();
+
+  const tint = theme.palette.surface?.tint;
+  const finalHeaderBackgroundColor =
+    headerBackgroundColor || tint || theme.palette.background.default;
+  const usingTint = !headerBackgroundColor && Boolean(tint);
+  const finalHeaderTextColor =
+    headerTextColor || (usingTint ? theme.palette.common.white : theme.palette.text.primary);
```
Comment (NSTKrishna) [inline src/custom/BottomSheet/BottomSheet.tsx]: it should be theme.palette.text.inverse : theme.palette.text.default

**Review by NSTKrishna — CHANGES_REQUESTED** [review]

**Review by NSTKrishna — APPROVED** [review]

**Review by PARTH-TUSSLE — CHANGES_REQUESTED** [review]
Thanks @KhushamBansal! The fallback to `background.default`, sharing the header foreground across the title/close icon, and prop forwarding in `DashboardLayout` look great.

One remaining contrast edge case: if someone passes a dark custom `headerBackgroundColor` (e.g. `#121212`), `usingTint` is `false`,
 which defaults to `text.default` (dark text in light mode), resulting in low contrast.

We can use Sistent's `readableTextColor` helper from `../../theme` so custom backgrounds adapt automatically while keeping explicit
`headerTextColor` as the priority override:

```tsx
    import { useTheme, readableTextColor } from '../../theme';

    // ...
    const tint = theme.palette.surface?.tint;
    const finalHeaderBackgroundColor =
      headerBackgroundColor ?? tint ?? theme.palette.background.default;

    const defaultForeground = headerBackgroundColor
      ? readableTextColor(headerBackgroundColor, theme.palette.text.inverse, theme.palette.text.default)
      : tint ? theme.palette.text.inverse : theme.palette.text.default;

    const finalHeaderTextColor = headerTextColor ?? defaultForeground;
```
Could you also add a few quick unit tests for BottomSheet covering:
  • Default surface.tint fallback
  • Missing tint → background.default fallback
  • Custom light / dark background contrast
  • Explicit headerTextColor override

  Other than that, this looks ready to go!

File: src/custom/DashboardLayout/DashboardLayout.tsx [hunk 3]
```diff
@@ -36,7 +42,9 @@ export const DashboardLayout: React.FC<DashboardLayoutProps> = ({
   sidebarTitle = 'Widget Picker',
   sidebarWidth = { xs: '100%', md: '350px' },
   sidebarTopOffset = '0',
-  sidebarHeight = '100vh'
+  sidebarHeight = '100vh',
```
Comment (Katotodan) [inline src/custom/DashboardLayout/DashboardLayout.tsx]: Would it be better to set `sidebarHeight` to `100dvh` instead of `100vh` to account for dynamic viewport changes?

**Review by Katotodan — COMMENTED** [review]
Looking good overall, just one suggestion.

**Review by ayanasarkar — COMMENTED** [review]
verified the color-resolution logic and ran the new test suite locally — all 7 pass, and the dark-mode tint/contrast edge case Parth flagged is correctly handled. Only open item is Katotodan's 100vh → 100dvh suggestion on sidebarHeight, which doesn't look addressed yet — otherwise this looks ready to merge once that's in.

File: src/custom/DashboardLayout/DashboardLayout.tsx (hunk 3 above)
Comment (KhushamBansal) [reply src/custom/DashboardLayout/DashboardLayout.tsx]: Fixed! Thanks for the review.

**Review by Katotodan — APPROVED** [review]

**Review by PARTH-TUSSLE — DISMISSED** [review]
Hey @KhushamBansal ,
1. The empty-string fix wasn't applied symmetrically. The background fallback was hardened to treat "" as absent, but the sibling text-color line still uses ?? So headerTextColor="" would still slip through as an invalid CSS color , silently reproducing the exact bug this PR's last content commit was written to fix , just on the other prop. Small, edge-case, but worth a one-line consistency fix (headerTextColor || defaultForeground) before merge.
2. No regression test for the empty-string case. The test suite (7 tests) is solid , tint fallback in both modes, background.default fallback, light/dark custom-background contrast, explicit override, icon color inheritance , but it doesn't cover the headerBackgroundColor="" scenario, nor the headerTextColor="" gap above. Worth adding one test for each so this doesn't regress silently later.

Other than that LGTM 👍

**Review by saurabhraghuvanshii — APPROVED** [review]
